### PR TITLE
fix/misc catchup problems

### PR DESCRIPTION
**Pull request by 0xArdi**
## Fixes

This PR fixes various issues found while testing rejoining of agents across multiple periods (reset indexes).
The issues that it fixes:
- Restarts tendermint whenever there's a connection drop, from the abci or otherwise.
- Makes sure monitoring is performed even when tendermint is reset by the tendermint server. Previous to this PR, we wouldn't see from tendermint if it had been reset due to a failure, like the ones noted above.
- Clean up timeouts and `last_timestamp` before trying to restore state received from peers.
- Don't wait for the observation interval in cases when reset is performed for recovery. This was the case before as well, now it's more explicit, and avoids running into `last_timestamp is None` errors when recovering.
- Enables ACN communication for rejoining agents, by accepting requests/responses from `all_participants`. 
- Fixes https://github.com/valory-xyz/open-autonomy/issues/1655.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [x] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have locally run services that could be impacted and they do not present failures derived from my changes

### CODE REVIEW

File: packages/valory/skills/abstract_round_abci/models.py [hunk 1]
```diff
@@ -532,6 +541,15 @@ class TendermintRecoveryParams(TypeCheckMixin):
     reset_params: Optional[List[Tuple[str, str]]] = None
     serialized_db_state: Optional[str] = None
 
+    def __hash__(self) -> int:
+        """Hash the object."""
+        return hash(
+            self.reset_from_round
+            + str(self.round_count)
+            + str(self.serialized_db_state)
+            + json.dumps(self.reset_params)
```
Comment (DavidMinarsch) [inline packages/valory/skills/abstract_round_abci/models.py:550]: Sort keys?

Comment (Adamantios) [reply packages/valory/skills/abstract_round_abci/models.py:550]: Not necessary for determinism since we are using python >= 3.7

Comment (0xArdi) [reply packages/valory/skills/abstract_round_abci/models.py:550]: This is a list of tuples actually, so that shouldnt make a difference.

Comment (DavidMinarsch) [reply packages/valory/skills/abstract_round_abci/models.py:550]: It's recursive afaik and we should always use it imo. Technically, there could be a non-Python client interacting. This just makes it all more transparent.

Comment (0xArdi) [reply packages/valory/skills/abstract_round_abci/models.py]: After merging `fix/api-specs` the `reset_params` is actually `Dict[str, str]`, `sort_keys=True` was added.

File: packages/valory/agents/register_reset/tests/test_register_reset.py [hunk 2]
```diff
@@ -56,7 +56,6 @@
 
 
 HAPPY_PATH = (
-    RoundChecks(RegistrationStartupRound.auto_round_id()),
```
[the remark this answers is not in the document]
Comment (0xArdi) [reply packages/valory/agents/register_reset/tests/test_register_reset.py:59]: Yes, the rejoining agent will not go through this.

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**